patroni/ha.py

```diff
@@ -733,8 +775,18 @@ def process_unhealthy_cluster(self):
                         logger.info('Cleaning up failover key after acquiring leader lock...')
                         self.dcs.manual_failover('', '')
                 self.load_cluster_from_dcs()
-                return self.enforce_master_role('acquired session lock as a leader',
-                                                'promoted self to leader by acquiring session lock')
+
+                if self.cluster.is_standby_cluster():
+                    # standby leader disappeared, and this is a healthiest
+                    # replica, so it should become a new standby leader.
+                    # This imply that we need to start following a remote master
+                    self.enforce_follow_remote_master()
```

Comment:
I'd pass the message at line 784 to the enforce_follow_remote_master and return its result, which should be either  a message that the actual Postgres master demoted, or that we have promoted ourself to the standby leader and following the remote leader. Right now, the message from line 448 is simply discarded here.